static/js/stream_data.js

```diff
@@ -557,6 +557,10 @@ export function can_subscribe_others(sub) {
 }
 
 export function can_post_messages_in_stream(stream) {
+    if (page_params.is_spectator) {
+        return false;
+    }
```

Comment:
I moved this check first, since it felt easier to read to not put this between the next two blocks.